packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  AzureSolutionSettings,
+  SolutionContext,
+  Plugin,
+  Result,
+  ok,
+  FxError,
+} from "@microsoft/teamsfx-api";
+import { ScaffoldArmTemplateResult } from "../../../common/armInterface";
+import { getActivatedResourcePlugins } from "./ResourcePluginContainer";
+import { getPluginContext } from "./util";
+import { format } from "util";
+import { compileHandlebarsTemplateString, getStrings } from "../../../common";
+import path from "path";
+import * as fs from "fs-extra";
+
+const baseFolder: string = "./infra/azure";
+const templateFolder: string = "templates";
+const parameterFolder: string = "parameters";
+const bicepOrchestrationFileName: string = "main.bicep";
+const parameterTemplateFileName: string = "parameter.template.json";
+const scaffoldArmTemplateInterfaceName: string = "scaffoldArmTemplate"; // Temporary solution before adding it to teamsfx-api
+const solutionLevelParameters: string = `param resourceBaseName string\n`;
+
+// Get ARM template content from each resource plugin and output to project folder
+export async function generateArmTemplate(ctx: SolutionContext): Promise<Result<any, FxError>> {
+  const azureSolutionSettings = ctx.projectSettings?.solutionSettings as AzureSolutionSettings;
+  const plugins = getActivatedResourcePlugins(azureSolutionSettings); // This function ensures return result won't be empty
+
+  let bicepOrchestrationTemplate = new BicepOrchestrationTemplate(plugins.map((p) => p.name));
+  let moduleFiles = new Map<string, string>();
+
+  // Get bicep content from each resource plugin
+  for (const plugin of plugins) {
+    //@ts-ignore temporary solution before adding related interface to teamsfx-api
+    if (plugin[scaffoldArmTemplateInterfaceName]) {
+      // find method using method name
+      const pluginContext = getPluginContext(ctx, plugin.name);
+      //@ts-ignore temporary solution before adding related interface to teamsfx-api
+      const result = (await plugin[scaffoldArmTemplateInterfaceName](pluginContext)) as Result<
+        ScaffoldArmTemplateResult,
+        FxError
+      >;
+      if (result.isOk()) {
+        bicepOrchestrationTemplate.addTemplate(plugin.name, result.value);
+        if (result.value.Modules) {
+          for (const module of Object.entries(result.value.Modules)) {
+            const moduleFileName = module[0];
+            const moduleFileContent = module[1].Content;
+            moduleFiles.set(generateBicepModuleFilePath(moduleFileName), moduleFileContent);
+          }
+        }
+      } else {
+        const msg = format(getStrings().solution.ProvisionFailNotice, ctx.projectSettings?.appName);
+        ctx.logProvider?.error(msg);
```

Comment:
the error in result is not output here?

Reply:
I'm following the practice in existing solution logic. Methods (scaffold/provision/etc.) in solution plugin returns the error result to core and core returns to VS Code Extension / TeamsFx CLI. It should be VS Code extension / TeamsFx CLI that finally shows the error to user.